src/Symfony/Component/Form/Extension/Core/Type/TextType.php

```diff
@@ -27,6 +27,9 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         // See https://github.com/symfony/symfony/issues/5906#issuecomment-203189375
         if ('' === $options['empty_data']) {
             $builder->addViewTransformer($this);
+        } elseif (FormType::$emptyData === $options['empty_data']) {
```

Comment:
At that point, this won't be the case, as the closure stored in `FormType::$emptyData` is the closure called internally in the OptionResolver, not the resolved option value.

You need to add one legacy test (per impacted type, maybe) asserting the deprecation. You will see that it fails currently.

Reply:
I fixed the problem and added a test to demonstrate the case. How can I ignore the deprecation in this test? I put `@group legacy` and does not help.

Reply:
See test `Symfony\Component\Form\Tests\Extension\Core\Type\TextTypeTest::testDefaultEmptyDataCallback()` for this case.